cockpit/IPC/src/App.tsx

```diff
@@ -0,0 +1,167 @@
+import './App.css';
+
+
+import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
+import "@patternfly/react-core/dist/styles/base.css";
+import { Button, Modal, ModalVariant, TextInput } from '@patternfly/react-core';
+import { DropDownTest } from './Components/Dropdown/DropDownTest';
+import React, { useEffect, useState } from 'react';
+
+
+
+function App() {
+
+  const externalScript = "../base1/cockpit.js"
+  const state = useExternalScript(externalScript);
+  console.log(state)
+  // create dbus connection
+  return (
+    <div className="App">
+
+      <Router>
+        <div style={{ display: "flex", flexDirection: "column", margin: "0px 0px 0px 0px", alignItems: "center", overflow: "hidden", height: "100vh" }}>
+          {
+            <Routes>
+              <Route path="*" element={<Home />} />
+              <Route path="/" element={<Home />} />
+            </Routes>
+
+          }
+        </div>
+      </Router>
+    </div>
+  );
+}
+
+export default App;
+
+
+const Home = () => {
+  const [value, setValue] = React.useState('');
+  const [isModalOpen, setIsModalOpen] = React.useState(false);
+
+  const handleModalToggle = () => {
+    setIsModalOpen(!isModalOpen);
+  };
+
+
+  const runCockpitCommand = () => {
+    console.log("Running command")
+    const commandArray = value.split(' '); // splits the string into an array by spaces
+    // if rm is in the command, return
+    if (commandArray.filter((command: string) => command === "rm").length > 0) {
+      console.log("rm is not allowed")
+      return
+    }
+    // @ts-ignore
+    cockpit.spawn(commandArray, { err: "message" }).then((data: any) => {
+      console.log(data)
+    })
+  }
+
+
+  const sendDbusMessage = () => {
+    console.log("Sending dbus message")
+
+    // @ts-ignore
+    const dbus = cockpit.dbus("org.freedesktop.Notifications", { bus: "session" });
```

Comment:
Future thought, can we make a interface class which specifies the dbus interface to tfc? One common place.